### PR TITLE
Connector builder api: create, update, delete

### PR DESCRIPTION
**Pull request by flash1293**
Based on https://github.com/airbytehq/airbyte/pull/22879

Fixes https://github.com/airbytehq/airbyte/issues/22570
Fixes https://github.com/airbytehq/airbyte/issues/22571
Fixes https://github.com/airbytehq/airbyte/issues/22572

This PR is adding the API layer for creating, updating and deleting builder projects based on the new model. This includes the following parts:
* Extend openapi spec for the server (`airbyte-api/src/main/openapi/config.yaml`)
* Implement API controller (`airbyte-server/src/main/java/io/airbyte/server/apis/ConnectorBuilderProjectApiController.java`) which is forwarding everything to the handler
* Implement handler (`airbyte-commons-server/src/main/java/io/airbyte/commons/server/handlers/ConnectorBuilderProjectsHandler.java`) which is executing the write operations with help of the `ConfigRepository`
  * Checking whether the project exists in the current workspace is centralized in a `validateWorkspace` method that check with the database
* Adjust config repository `airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/ConfigRepository.java`:
  * Do not check workspace consistency (this is pare of the handler now)
  * Throw instead of returning optional for the builder project for consistency
* Add tests for the handler and adjust tests for the config repository.

Notes:
* I tried to follow established approaches whereever possible to make this code as little "suprising" as possible for a platform engineer, but maybe I misunderstood something
* In theory it would be possible in some places to collapse multiple queries into one (e.g. the validate call is already implicitly checking whether a builder project exists, but the write call is doing another lightweight exists call), however separating different concerns seemed more important here. If it turns out to be necessary, performance optimizations can be done in a later step.

Can be tested end-to-end by building the server via `SUB_BUILD=PLATFORM ./gradlew assemble`, starting via `BASIC_AUTH_USERNAME="" BASIC_AUTH_PASSWORD="" VERSION=dev docker-compose --file ./docker-compose.yaml up` and using curl to call the new endpoints while checking the state of the database by logging into the database via `docker exec -ti airbyte-db psql -U docker -d airbyte`

### CODE REVIEW

**Review by xiaohansong — APPROVED** [review]

File: airbyte-commons-server/src/main/java/io/airbyte/commons/server/handlers/ConnectorBuilderProjectsHandler.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+import io.airbyte.config.persistence.ConfigNotFoundException;
+import io.airbyte.config.persistence.ConfigRepository;
+import io.airbyte.validation.json.JsonValidationException;
+import jakarta.inject.Inject;
+import jakarta.inject.Singleton;
+import java.io.IOException;
+import java.util.UUID;
+import java.util.function.Supplier;
+
+@SuppressWarnings("PMD.AvoidCatchingNPE")
```
Comment (girarda) [inline airbyte-commons-server/src/main/java/io/airbyte/commons/server/handlers/ConnectorBuilderProjectsHandler.java:22]: why suppress the warning?

File: airbyte-commons-server/src/main/java/io/airbyte/commons/server/handlers/ConnectorBuilderProjectsHandler.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    return new ConnectorBuilderProject().withBuilderProjectId(id).withWorkspaceId(projectCreate.getWorkspaceId())
+        .withName(projectCreate.getBuilderProject().getName())
+        .withManifestDraft(new ObjectMapper().valueToTree(projectCreate.getBuilderProject().getDraftManifest()));
+  }
+
+  private ConnectorBuilderProjectIdWithWorkspaceId idResponseFromBuilderProject(final ConnectorBuilderProject project) {
+    return new ConnectorBuilderProjectIdWithWorkspaceId().workspaceId(project.getWorkspaceId()).builderProjectId(project.getBuilderProjectId());
+  }
+
+  public void validateWorkspace(final UUID projectId, final UUID workspaceId) throws ConfigNotFoundException, IOException {
```
Comment (girarda) [inline airbyte-commons-server/src/main/java/io/airbyte/commons/server/handlers/ConnectorBuilderProjectsHandler.java:61]: can this method be package-protected? I don't think it should be used directly from another class

**Review by girarda — APPROVED** [review]

File: airbyte-commons-server/src/main/java/io/airbyte/commons/server/handlers/ConnectorBuilderProjectsHandler.java (hunk 1 above)
Comment (flash1293) [reply airbyte-commons-server/src/main/java/io/airbyte/commons/server/handlers/ConnectorBuilderProjectsHandler.java:22]: Leftover, removed that.

File: airbyte-commons-server/src/main/java/io/airbyte/commons/server/handlers/ConnectorBuilderProjectsHandler.java (hunk 2 above)
Comment (flash1293) [reply airbyte-commons-server/src/main/java/io/airbyte/commons/server/handlers/ConnectorBuilderProjectsHandler.java:61]: You are right, made it private

**Pull request closed without merging**